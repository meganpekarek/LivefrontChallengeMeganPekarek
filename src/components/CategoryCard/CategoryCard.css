.categoryCard__wrapper {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    border-radius: .4rem;
    padding: 2rem;
    background: var(--color-background);
    width: 20rem;
    height: 20rem;
    box-shadow: rgba(0, 0, 0, 0.2) 0px 2px 6px;
    cursor: pointer;
}

.categoryCard__wrapper:hover {
    background: var(--color-background-hover);
}

.categoryCard__header {
    font-family: var(--font-family-primary);
    text-align: center;
}

.categoryCard__desc {
    font-family: var(--font-family-secondary);
    font-weight: 700;
    font-size: 1.4rem;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

/* .categoryCard__wrapper:hover {
    height: auto;
    white-space: unset;
} */

.categoryCard__imgWrapper {
    display: flex;
    justify-content: center;
}

.categoryCard__img {
    width: 12rem;
}