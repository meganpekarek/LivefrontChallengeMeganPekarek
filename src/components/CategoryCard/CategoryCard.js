import './CategoryCard.css';
import React from 'react'

function CategoryCard(props) {
  const {category} = props;

  const goToRecipes = () => {
    props.history.push({
      pathname: '/recipes',
      state: {
        category: category
      }
    })
  }

  return (
      <React.Fragment>
        <div className="categoryCard__wrapper" onClick={goToRecipes}>
          <div className="categoryCard__imgWrapper">
            <img className="categoryCard__img" src={category.strCategoryThumb} alt="food-category" />
          </div>
          <div>
            <header className="categoryCard__header">{category.strCategory}</header>
          </div>
        </div>
      </React.Fragment>
  );
}

export default CategoryCard;
