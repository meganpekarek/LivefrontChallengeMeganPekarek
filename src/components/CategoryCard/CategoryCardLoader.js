import Skeleton, { SkeletonTheme } from "react-loading-skeleton";
import React from 'react'
import './CategoryCard.css';

function CategoryCardLoader(props) {

    return (
        <SkeletonTheme color="white" highlightColor="lightgrey">
            <div className="categoryCard__wrapper">
                <div className="categoryCard__imgWrapper">
                    <Skeleton circle={true} height={70} width={70} />
                </div>
                <div>
                    <Skeleton className="categoryCard__header" />
                </div>
            </div>
        </SkeletonTheme>
    );
}

export default CategoryCardLoader;
