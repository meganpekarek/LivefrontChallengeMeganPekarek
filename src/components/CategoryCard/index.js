import CategoryCard from "./CategoryCard.js";

export default CategoryCard;
