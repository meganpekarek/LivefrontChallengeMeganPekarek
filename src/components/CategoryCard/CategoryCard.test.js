import React from 'react';
import renderer from 'react-test-renderer';
import CategoryCard from './CategoryCard';

test('Card color changes on hover', () => {
  const categoryObj = {
      strCategory: 'testCat',
      strCategoryDescription: 'description of testCat',
      strCategoryThumb: 'https://www.themealdb.com/images/media/meals/wrssvt1511556563.jpg'
  }
  const component = renderer.create(
    <CategoryCard category={categoryObj}></CategoryCard>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  console.log(tree)
  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});