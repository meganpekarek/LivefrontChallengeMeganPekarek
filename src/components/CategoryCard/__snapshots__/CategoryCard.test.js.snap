// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card color changes on hover 1`] = `
<div
  className="categoryCard__wrapper"
  onClick={[Function]}
>
  <div
    className="categoryCard__imgWrapper"
  >
    <img
      alt="food-category"
      className="categoryCard__img"
      src="https://www.themealdb.com/images/media/meals/wrssvt1511556563.jpg"
    />
  </div>
  <div>
    <header
      className="categoryCard__header"
    >
      testCat
    </header>
    <p
      className="categoryCard__desc"
    >
      description of testCat
    </p>
  </div>
</div>
`;
